### PR TITLE
Fix crash in designer discovery

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
This regressed with https://github.com/dotnet/roslyn/pull/71999 because we changed the logic from: "sync solution (which only syncs C#/VB projects), then process whole solution on remote site" to "first, attempt to analyze the active document, then do the original full-sync and analyze".  However, the code to analyze the active document was invalid for the case of this not being a C#/VB file.  IN that case, we'd sync over the C#/VB subset of projects, then not find the non-C#/VB file on the oop side, immediately crashing.

### CODE REVIEW

File: src/Workspaces/Core/Portable/Remote/RemoteSupportedLanguages.cs [hunk 1]
```diff
@@ -2,16 +2,12 @@
...
-    internal static class RemoteSupportedLanguages
-    {
-        public static bool IsSupported(this string language)
-        {
-            return language is LanguageNames.CSharp or
-                   LanguageNames.VisualBasic;
-        }
-    }
+    public static bool IsSupported([NotNullWhen(true)] this string? language)
+        => language is LanguageNames.CSharp or LanguageNames.VisualBasic;
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Remote/RemoteSupportedLanguages.cs:12]: I never like these null-accepting things like this. :cry: (Not blocking but I can still be sad in my own corner.)

File: src/VisualStudio/Core/Def/DesignerAttribute/VisualStudioDesignerAttributeService.cs [hunk 2]
```diff
@@ -122,14 +122,16 @@ private async ValueTask ProcessWorkspaceChangeAsync(CancellationToken cancellati
                 return;
 
             var trackingService = _workspace.Services.GetRequiredService<IDocumentTrackingService>();
-            var priorityDocumentId = trackingService.GetActiveDocument(solution)?.Id;
+            var activeDocument = trackingService.GetActiveDocument(solution);
 
             // If there is an active document, then process changes to it right away, so that the UI updates quickly
             // when the user adds/removes a form from a particular document.
-            if (priorityDocumentId != null)
+            if (RemoteSupportedLanguages.IsSupported(activeDocument?.Project.Language))
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/DesignerAttribute/VisualStudioDesignerAttributeService.cs]: Can we easily add a test for this?

**Review by jasonmalinowski — APPROVED** [review]
Signing off; it'd be good to have a test for this but merging is fine to get stuff unblocked.

**Review by arkalyanms — APPROVED** [review]

File: src/VisualStudio/Core/Def/DesignerAttribute/VisualStudioDesignerAttributeService.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/DesignerAttribute/VisualStudioDesignerAttributeService.cs]: workign on test now.

**Pull request merged**